### ISSUE TITLE
[Bug]: MicrosoftWvdRemoteDesktop downloads moved?

### ISSUE THREAD

**Issue by iainbrighton** [aaronparker/evergreen#352]
### What happened?

It looks like MS have updated some of the download links for the AVD client. Updating the manifest with the [new links](https://docs.microsoft.com/en-us/azure/virtual-desktop/connect-windows-7-10) works, but I can't find the updated Insider links.

```
[20] D:\Scratch> Get-EvergreenApp -Name MicrosoftWvdRemoteDesktop
WARNING: Invoke-WebRequestWrapper: Error at URI: https://go.microsoft.com/fwlink/?linkid=2139368.
WARNING: Invoke-WebRequestWrapper: Error encountered: The remote server returned an error: (503) Server Unavailable..
WARNING: Invoke-WebRequestWrapper: For troubleshooting steps see: https://stealthpuppy.com/evergreen/troubleshoot/.
Invoke-WebRequestWrapper : Invoke-WebRequestWrapper: The remote server returned an error: (503) Server Unavailable..
At C:\Users\IainBrighton\OneDrive - Virtual Engine
Limited\PowerShell\Modules\Evergreen\Evergreen\Apps\Get-MicrosoftWvdRemoteDesktop.ps1:32 char:24
+             $Headers = Invoke-WebRequestWrapper @params
+                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,Invoke-WebRequestWrapper

WARNING: Invoke-WebRequestWrapper: Error at URI: https://go.microsoft.com/fwlink/?linkid=2139144.
WARNING: Invoke-WebRequestWrapper: Error encountered: The remote server returned an error: (503) Server Unavailable..
WARNING: Invoke-WebRequestWrapper: For troubleshooting steps see: https://stealthpuppy.com/evergreen/troubleshoot/.
Invoke-WebRequestWrapper : Invoke-WebRequestWrapper: The remote server returned an error: (503) Server Unavailable..
At C:\Users\IainBrighton\OneDrive - Virtual Engine
Limited\PowerShell\Modules\Evergreen\Evergreen\Apps\Get-MicrosoftWvdRemoteDesktop.ps1:32 char:24
+             $Headers = Invoke-WebRequestWrapper @params
+                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,Invoke-WebRequestWrapper

WARNING: Invoke-WebRequestWrapper: Error at URI: https://go.microsoft.com/fwlink/?linkid=2139368.
WARNING: Invoke-WebRequestWrapper: Error encountered: The remote server returned an error: (503) Server Unavailable..
WARNING: Invoke-WebRequestWrapper: For troubleshooting steps see: https://stealthpuppy.com/evergreen/troubleshoot/.
Invoke-WebRequestWrapper : Invoke-WebRequestWrapper: The remote server returned an error: (503) Server Unavailable..
At C:\Users\IainBrighton\OneDrive - Virtual Engine
Limited\PowerShell\Modules\Evergreen\Evergreen\Apps\Get-MicrosoftWvdRemoteDesktop.ps1:32 char:24
+             $Headers = Invoke-WebRequestWrapper @params
+                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,Invoke-WebRequestWrapper

WARNING: Invoke-WebRequestWrapper: Error at URI: https://go.microsoft.com/fwlink/?linkid=2139144.
WARNING: Invoke-WebRequestWrapper: Error encountered: The remote server returned an error: (503) Server Unavailable..
WARNING: Invoke-WebRequestWrapper: For troubleshooting steps see: https://stealthpuppy.com/evergreen/troubleshoot/.
Invoke-WebRequestWrapper : Invoke-WebRequestWrapper: The remote server returned an error: (503) Server Unavailable..
At C:\Users\IainBrighton\OneDrive - Virtual Engine
Limited\PowerShell\Modules\Evergreen\Evergreen\Apps\Get-MicrosoftWvdRemoteDesktop.ps1:32 char:24
+             $Headers = Invoke-WebRequestWrapper @params
+                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,Invoke-WebRequestWrapper



Version      : 1.2.3313.0
Architecture : x64
Channel      : Insider
Date         : 23/06/2022
Size         : 25206784
Filename     : RemoteDesktop_1.2.3313.0_x64.msi
URI          : https://query.prod.cms.rt.microsoft.com/cms/api/am/binary/RE50t3Q

Version      : 1.2.3213.0
Architecture : x64
Channel      : Public
Date         : 01/06/2022
Size         : 24444928
Filename     : RemoteDesktop_1.2.3213.0_x64.msi
URI          : https://query.prod.cms.rt.microsoft.com/cms/api/am/binary/RE4YfFO
```

### Version

2206.583

### What PowerShell edition/s are you running Evergreen on?

Windows PowerShell

### Which operating system/s are you running Evergreen on?

Windows 10

### Have you reviewed the documentation?

- [X] Troubleshooting at: https://stealthpuppy.com/evergreen/troubleshoot/
- [X] Known issues at: https://stealthpuppy.com/evergreen/issues/

### Verbose output

```shell
[27] D:\Scratch> Get-EvergreenApp -Name MicrosoftWvdRemoteDesktop -Verbose
VERBOSE: Get-EvergreenApp: Function exists: C:\Users\IainBrighton\OneDrive - Virtual Engine
Limited\PowerShell\Modules\Evergreen\Evergreen\Apps\Get-MicrosoftWvdRemoteDesktop.ps1.
VERBOSE: Get-EvergreenApp: Dot sourcing: C:\Users\IainBrighton\OneDrive - Virtual Engine
Limited\PowerShell\Modules\Evergreen\Evergreen\Apps\Get-MicrosoftWvdRemoteDesktop.ps1.
VERBOSE: Get-FunctionResource: read application resource strings from [C:\Users\IainBrighton\OneDrive - Virtual Engine
Limited\PowerShell\Modules\Evergreen\Evergreen\Manifests\MicrosoftWvdRemoteDesktop.json]
VERBOSE: Get-EvergreenApp: Calling: Get-MicrosoftWvdRemoteDesktop.
VERBOSE: Get-MicrosoftWvdRemoteDesktop: Querying for channel: Public.
VERBOSE: Get-MicrosoftWvdRemoteDesktop: Querying for architecture: ARM64.
VERBOSE: Invoke-WebRequestWrapper: Invoke-WebRequest parameter: [UserAgent: Mozilla/5.0 (Windows NT; Windows NT 10.0;
en-GB) AppleWebKit/534.6 (KHTML, like Gecko) Chrome/7.0.500.0 Safari/534.6].
VERBOSE: Invoke-WebRequestWrapper: Invoke-WebRequest parameter: [Method: Head].
VERBOSE: Invoke-WebRequestWrapper: Invoke-WebRequest parameter: [ErrorAction: Continue].
VERBOSE: Invoke-WebRequestWrapper: Invoke-WebRequest parameter: [UseBasicParsing: True].
VERBOSE: Invoke-WebRequestWrapper: Invoke-WebRequest parameter: [Uri: https://go.microsoft.com/fwlink/?linkid=2139368].
VERBOSE: HEAD with 0-byte payload
WARNING: Invoke-WebRequestWrapper: Error at URI: https://go.microsoft.com/fwlink/?linkid=2139368.
WARNING: Invoke-WebRequestWrapper: Error encountered: The remote server returned an error: (503) Server Unavailable..
WARNING: Invoke-WebRequestWrapper: For troubleshooting steps see: https://stealthpuppy.com/evergreen/troubleshoot/.
Invoke-WebRequestWrapper : Invoke-WebRequestWrapper: The remote server returned an error: (503) Server Unavailable..
At C:\Users\IainBrighton\OneDrive - Virtual Engine
Limited\PowerShell\Modules\Evergreen\Evergreen\Apps\Get-MicrosoftWvdRemoteDesktop.ps1:32 char:24
+             $Headers = Invoke-WebRequestWrapper @params
+                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,Invoke-WebRequestWrapper

VERBOSE: Get-MicrosoftWvdRemoteDesktop: Querying for architecture: x86.
VERBOSE: Invoke-WebRequestWrapper: Invoke-WebRequest parameter: [UserAgent: Mozilla/5.0 (Windows NT; Windows NT 10.0;
en-GB) AppleWebKit/534.6 (KHTML, like Gecko) Chrome/7.0.500.0 Safari/534.6].
VERBOSE: Invoke-WebRequestWrapper: Invoke-WebRequest parameter: [Method: Head].
VERBOSE: Invoke-WebRequestWrapper: Invoke-WebRequest parameter: [ErrorAction: Continue].
VERBOSE: Invoke-WebRequestWrapper: Invoke-WebRequest parameter: [UseBasicParsing: True].
VERBOSE: Invoke-WebRequestWrapper: Invoke-WebRequest parameter: [Uri: https://go.microsoft.com/fwlink/?linkid=2139144].
VERBOSE: HEAD with 0-byte payload
WARNING: Invoke-WebRequestWrapper: Error at URI: https://go.microsoft.com/fwlink/?linkid=2139144.
WARNING: Invoke-WebRequestWrapper: Error encountered: The remote server returned an error: (503) Server Unavailable..
WARNING: Invoke-WebRequestWrapper: For troubleshooting steps see: https://stealthpuppy.com/evergreen/troubleshoot/.
Invoke-WebRequestWrapper : Invoke-WebRequestWrapper: The remote server returned an error: (503) Server Unavailable..
At C:\Users\IainBrighton\OneDrive - Virtual Engine
Limited\PowerShell\Modules\Evergreen\Evergreen\Apps\Get-MicrosoftWvdRemoteDesktop.ps1:32 char:24
+             $Headers = Invoke-WebRequestWrapper @params
+                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,Invoke-WebRequestWrapper

VERBOSE: Get-MicrosoftWvdRemoteDesktop: Querying for architecture: x64.
VERBOSE: Invoke-WebRequestWrapper: Invoke-WebRequest parameter: [UserAgent: Mozilla/5.0 (Windows NT; Windows NT 10.0;
en-GB) AppleWebKit/534.6 (KHTML, like Gecko) Chrome/7.0.500.0 Safari/534.6].
VERBOSE: Invoke-WebRequestWrapper: Invoke-WebRequest parameter: [Method: Head].
VERBOSE: Invoke-WebRequestWrapper: Invoke-WebRequest parameter: [ErrorAction: Continue].
VERBOSE: Invoke-WebRequestWrapper: Invoke-WebRequest parameter: [UseBasicParsing: True].
VERBOSE: Invoke-WebRequestWrapper: Invoke-WebRequest parameter: [Uri: https://go.microsoft.com/fwlink/?linkid=2139369].
VERBOSE: HEAD with 0-byte payload
VERBOSE: received 24444928-byte response of content type application/x-msdownload
VERBOSE: Invoke-WebRequestWrapper: Response: [200].
VERBOSE: Invoke-WebRequestWrapper: Content type: [application/x-msdownload].
VERBOSE: Invoke-WebRequestWrapper: Returning headers.
VERBOSE: ConvertTo-DateTime: Converting: [Wed, 01 Jun 2022 22:20:27 GMT].
VERBOSE: ConvertTo-DateTime: Attempting to convert date format to: en-GB.
VERBOSE: ConvertTo-DateTime: Returning date: [01/06/2022].
VERBOSE: Get-MicrosoftWvdRemoteDesktop: Querying for channel: Insider.
VERBOSE: Get-MicrosoftWvdRemoteDesktop: Querying for architecture: ARM64.
VERBOSE: Invoke-WebRequestWrapper: Invoke-WebRequest parameter: [UserAgent: Mozilla/5.0 (Windows NT; Windows NT 10.0;
en-GB) AppleWebKit/534.6 (KHTML, like Gecko) Chrome/7.0.500.0 Safari/534.6].
VERBOSE: Invoke-WebRequestWrapper: Invoke-WebRequest parameter: [Method: Head].
VERBOSE: Invoke-WebRequestWrapper: Invoke-WebRequest parameter: [ErrorAction: Continue].
VERBOSE: Invoke-WebRequestWrapper: Invoke-WebRequest parameter: [UseBasicParsing: True].
VERBOSE: Invoke-WebRequestWrapper: Invoke-WebRequest parameter: [Uri: https://go.microsoft.com/fwlink/?linkid=2139368].
VERBOSE: HEAD with 0-byte payload
WARNING: Invoke-WebRequestWrapper: Error at URI: https://go.microsoft.com/fwlink/?linkid=2139368.
WARNING: Invoke-WebRequestWrapper: Error encountered: The remote server returned an error: (503) Server Unavailable..
WARNING: Invoke-WebRequestWrapper: For troubleshooting steps see: https://stealthpuppy.com/evergreen/troubleshoot/.
Invoke-WebRequestWrapper : Invoke-WebRequestWrapper: The remote server returned an error: (503) Server Unavailable..
At C:\Users\IainBrighton\OneDrive - Virtual Engine
Limited\PowerShell\Modules\Evergreen\Evergreen\Apps\Get-MicrosoftWvdRemoteDesktop.ps1:32 char:24
+             $Headers = Invoke-WebRequestWrapper @params
+                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,Invoke-WebRequestWrapper

VERBOSE: Get-MicrosoftWvdRemoteDesktop: Querying for architecture: x86.
VERBOSE: Invoke-WebRequestWrapper: Invoke-WebRequest parameter: [UserAgent: Mozilla/5.0 (Windows NT; Windows NT 10.0;
en-GB) AppleWebKit/534.6 (KHTML, like Gecko) Chrome/7.0.500.0 Safari/534.6].
VERBOSE: Invoke-WebRequestWrapper: Invoke-WebRequest parameter: [Method: Head].
VERBOSE: Invoke-WebRequestWrapper: Invoke-WebRequest parameter: [ErrorAction: Continue].
VERBOSE: Invoke-WebRequestWrapper: Invoke-WebRequest parameter: [UseBasicParsing: True].
VERBOSE: Invoke-WebRequestWrapper: Invoke-WebRequest parameter: [Uri: https://go.microsoft.com/fwlink/?linkid=2139144].
VERBOSE: HEAD with 0-byte payload
WARNING: Invoke-WebRequestWrapper: Error at URI: https://go.microsoft.com/fwlink/?linkid=2139144.
WARNING: Invoke-WebRequestWrapper: Error encountered: The remote server returned an error: (503) Server Unavailable..
WARNING: Invoke-WebRequestWrapper: For troubleshooting steps see: https://stealthpuppy.com/evergreen/troubleshoot/.
Invoke-WebRequestWrapper : Invoke-WebRequestWrapper: The remote server returned an error: (503) Server Unavailable..
At C:\Users\IainBrighton\OneDrive - Virtual Engine
Limited\PowerShell\Modules\Evergreen\Evergreen\Apps\Get-MicrosoftWvdRemoteDesktop.ps1:32 char:24
+             $Headers = Invoke-WebRequestWrapper @params
+                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,Invoke-WebRequestWrapper

VERBOSE: Get-MicrosoftWvdRemoteDesktop: Querying for architecture: x64.
VERBOSE: Invoke-WebRequestWrapper: Invoke-WebRequest parameter: [UserAgent: Mozilla/5.0 (Windows NT; Windows NT 10.0;
en-GB) AppleWebKit/534.6 (KHTML, like Gecko) Chrome/7.0.500.0 Safari/534.6].
VERBOSE: Invoke-WebRequestWrapper: Invoke-WebRequest parameter: [Method: Head].
VERBOSE: Invoke-WebRequestWrapper: Invoke-WebRequest parameter: [ErrorAction: Continue].
VERBOSE: Invoke-WebRequestWrapper: Invoke-WebRequest parameter: [UseBasicParsing: True].
VERBOSE: Invoke-WebRequestWrapper: Invoke-WebRequest parameter: [Uri: https://go.microsoft.com/fwlink/?linkid=2139233].
VERBOSE: HEAD with 0-byte payload
VERBOSE: received 25206784-byte response of content type application/x-msdownload
VERBOSE: Invoke-WebRequestWrapper: Response: [200].
VERBOSE: Invoke-WebRequestWrapper: Content type: [application/x-msdownload].
VERBOSE: Invoke-WebRequestWrapper: Returning headers.
VERBOSE: ConvertTo-DateTime: Converting: [Thu, 23 Jun 2022 16:23:39 GMT].
VERBOSE: ConvertTo-DateTime: Attempting to convert date format to: en-GB.
VERBOSE: ConvertTo-DateTime: Returning date: [23/06/2022].
VERBOSE: Get-EvergreenApp: Output result from: C:\Users\IainBrighton\OneDrive - Virtual Engine
Limited\PowerShell\Modules\Evergreen\Evergreen\Apps\Get-MicrosoftWvdRemoteDesktop.ps1.


Version      : 1.2.3313.0
Architecture : x64
Channel      : Insider
Date         : 23/06/2022
Size         : 25206784
Filename     : RemoteDesktop_1.2.3313.0_x64.msi
URI          : https://query.prod.cms.rt.microsoft.com/cms/api/am/binary/RE50t3Q

Version      : 1.2.3213.0
Architecture : x64
Channel      : Public
Date         : 01/06/2022
Size         : 24444928
Filename     : RemoteDesktop_1.2.3213.0_x64.msi
URI          : https://query.prod.cms.rt.microsoft.com/cms/api/am/binary/RE4YfFO
```

**Comment by aaronparker** [comment]
Already addressed here: 768f13d26a5f302b3cdaff973a43cb3f94ecea4a, but need to push this to a release version.

I'm actually seeing the original links provide inconsistent results, as in sometimes they work, some time they don't.

**Comment by aaronparker** [comment]
You can see the changes here if you view the page a few times a day (it update every 4 hours): https://stealthpuppy.com/apptracker/#microsoftwvdremotedesktop

**Comment by iainbrighton** [comment]
> Already addressed here: [768f13d](https://github.com/aaronparker/evergreen/commit/768f13d26a5f302b3cdaff973a43cb3f94ecea4a), but need to push this to a release version.
> 
> I'm actually seeing the original links provide inconsistent results, as in sometimes they work, some time they don't.

Ah - I did check the Issues log before logging it. Hopefully, the new links prove to be more reliable. God bless MS.

**Comment by aaronparker** [comment]
I'm still getting errors with the new URLs, so not sure what's going on here:

```powershell
PS /Users/aaron/projects/evergreen> $r = gea MicrosoftWvdRemoteDesktop
WARNING: Invoke-WebRequestWrapper: Error at URI: https://go.microsoft.com/fwlink/?linkid=2098961.
WARNING: Invoke-WebRequestWrapper: Error encountered: Response status code does not indicate success: 503 (Service Unavailable)..
WARNING: Invoke-WebRequestWrapper: For troubleshooting steps see: https://stealthpuppy.com/evergreen/troubleshoot/.
Write-Error: /Users/aaron/projects/evergreen/Evergreen/Apps/Get-MicrosoftWvdRemoteDesktop.ps1:32
Line |
  32 |              $Headers = Invoke-WebRequestWrapper @params
     |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Invoke-WebRequestWrapper: Response status code does not indicate success: 503 (Service Unavailable)..

WARNING: Invoke-WebRequestWrapper: Error at URI: https://go.microsoft.com/fwlink/?linkid=2098960.
WARNING: Invoke-WebRequestWrapper: Error encountered: Response status code does not indicate success: 503 (Service Unavailable)..
WARNING: Invoke-WebRequestWrapper: For troubleshooting steps see: https://stealthpuppy.com/evergreen/troubleshoot/.
Write-Error: /Users/aaron/projects/evergreen/Evergreen/Apps/Get-MicrosoftWvdRemoteDesktop.ps1:32
Line |
  32 |              $Headers = Invoke-WebRequestWrapper @params
     |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Invoke-WebRequestWrapper: Response status code does not indicate success: 503 (Service Unavailable)..

WARNING: Invoke-WebRequestWrapper: Error at URI: https://go.microsoft.com/fwlink/?linkid=2139368.
WARNING: Invoke-WebRequestWrapper: Error encountered: Response status code does not indicate success: 503 (Service Unavailable)..
WARNING: Invoke-WebRequestWrapper: For troubleshooting steps see: https://stealthpuppy.com/evergreen/troubleshoot/.
Write-Error: /Users/aaron/projects/evergreen/Evergreen/Apps/Get-MicrosoftWvdRemoteDesktop.ps1:32
Line |
  32 |              $Headers = Invoke-WebRequestWrapper @params
     |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Invoke-WebRequestWrapper: Response status code does not indicate success: 503 (Service Unavailable)..

WARNING: Invoke-WebRequestWrapper: Error at URI: https://go.microsoft.com/fwlink/?linkid=2139144.
WARNING: Invoke-WebRequestWrapper: Error encountered: Response status code does not indicate success: 503 (Service Unavailable)..
WARNING: Invoke-WebRequestWrapper: For troubleshooting steps see: https://stealthpuppy.com/evergreen/troubleshoot/.
Write-Error: /Users/aaron/projects/evergreen/Evergreen/Apps/Get-MicrosoftWvdRemoteDesktop.ps1:32
Line |
  32 |              $Headers = Invoke-WebRequestWrapper @params
     |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Invoke-WebRequestWrapper: Response status code does not indicate success: 503 (Service Unavailable)..
```

**Comment by aaronparker** [comment]
I've pushed a change to the development branch with some major changes, but there's still occasional issues when querying the download URLs to retrieve filenames.

**Comment by aaronparker** [comment]
Update to this released in `2207.592`, but those URLs still return 503 occasionally
